src/lib/pubkey/x25519/donna.cpp

```diff
@@ -376,14 +377,14 @@ void cmult(uint64_t resultx[5], uint64_t resultz[5], const uint8_t n[32], const
    copy_mem(a, q, 5);
 
    for(size_t i = 0; i < 32; ++i) {
-      const uint64_t bit0 = (n[31 - i] >> 7) & 1;
-      const uint64_t bit1 = (n[31 - i] >> 6) & 1;
-      const uint64_t bit2 = (n[31 - i] >> 5) & 1;
-      const uint64_t bit3 = (n[31 - i] >> 4) & 1;
-      const uint64_t bit4 = (n[31 - i] >> 3) & 1;
-      const uint64_t bit5 = (n[31 - i] >> 2) & 1;
-      const uint64_t bit6 = (n[31 - i] >> 1) & 1;
-      const uint64_t bit7 = (n[31 - i] >> 0) & 1;
+      const uint64_t bit0 = CT::value_barrier((n[31 - i] >> 7) & 1);
+      const uint64_t bit1 = CT::value_barrier((n[31 - i] >> 6) & 1);
+      const uint64_t bit2 = CT::value_barrier((n[31 - i] >> 5) & 1);
+      const uint64_t bit3 = CT::value_barrier((n[31 - i] >> 4) & 1);
+      const uint64_t bit4 = CT::value_barrier((n[31 - i] >> 3) & 1);
+      const uint64_t bit5 = CT::value_barrier((n[31 - i] >> 2) & 1);
+      const uint64_t bit6 = CT::value_barrier((n[31 - i] >> 1) & 1);
+      const uint64_t bit7 = CT::value_barrier((n[31 - i] >> 0) & 1);
```

Comment:
Perhaps worth adding a higher-level `CT::is_even()` for this (if it doesn't exist; I didn't check), to make this:
`const CT::Choice bit0 = CT::is_odd(n[31 - i] >> 7)`. This would also play nicely with my other comment.

Reply:
Already existing was `CT::Mask::expand_bit` which I used here.